### DISCUSSION TITLE
Not all tumblr images are downloaded within a blog

### DISCUSSION THREAD

**Discussion by Lumerica** [TumblThreeApp/TumblThree#312] [Q&A (Question & Answer)]
**Describe the bug**
Hello, I tried to download several tumblr blogs, it goes smoothly.
I have enabled the settings shown in the screenshots attached.

But when I check the downloaded files; I see many files that are not downloaded but the files does exist.
For example I see many json files with links to photos that are not downloaded. I have included 1 json file renamed the extension to txt.
In line 55 you can see that there is a link to a photo. However the app did not download neither the smaller size, thumbnail nor larger image; only the json file is downloaded.

**Desktop (please complete the following information):**
 - TumblThree version: [2.5.4.278]
 - OS: Windows 10
 - Browser [Firefox (version 97.0.1), Chrome (version 98.0.4758.102)]

**Additional context**
App settings:
![App_settings_1](https://user-images.githubusercontent.com/41524566/155909846-3ed7fa06-7c2c-4813-82bf-b884fe1577e8.png)
![App_settings_2](https://user-images.githubusercontent.com/41524566/155909858-a65821c0-d4dd-4382-a3f1-97ff4f034a35.png)

Blog setings:

1 of the many json file where only the json is downloaded containing the link:
[tumblr_mlqeauY3yC1r3k0o9o1_1280.json.txt](https://github.com/TumblThreeApp/TumblThree/files/8149912/tumblr_mlqeauY3yC1r3k0o9o1_1280.json.txt)

**Comment by thomas694** [comment]
Hello,
thanks for filing an issue report and providing much details.

But to be honest, the blog is downloading here, especially the mentioned images. So it seems to be some other problem. The mentioned images are quite at the start of the blog, so you shouldn't have hit a limit yet, except you already downloaded dozens of blogs before.
In general, after a full download without errors, 'LastId' inside the blog file should contain the number of the newest post. In case of errors, it remains 0 or the old one. What's the value in your file? You can also have a look in the `TumblThree.log` to check, if it contains errors like timeouts and others.
Did you download similar blogs before which maybe reposted these posts and you have the setting `check for file existence globally...` activated? Then these images aren't downloaded again, if already downloaded in another blog before.

You can test the blog again. Make sure you haven't already downloaded much at that moment. Set the download location in the settings to a completely new folder (you can go back to your old folder afterwards). Add this blog again, it should be the only one now. Check the (blog) settings and download the blog. No errors in the app should be shown and hardly errors in the log file should appear. Do the images download now?

**Comment by Lumerica** [comment]
Hello,

Thank you so much for your reply, really appreciate your swift response.

I will test it tomorrow just to make sure i am not locked out because of the limit. as I did download dozen before. But strange thing is that many posts does get downloaded.

Do you mind telling me where i can find this blog file with the 'LastId'?

You mentioned about the file 'TumblThree.log' I am not sure where this file is located at, as I doubt you are referring to the file located here: C:\users\<user>\appdata\Local\TumblThree\TumblThree.log
Not much is written in that file even with log mode set to Verbose.

By the way, before I download the blog again, I should just remove it from the list or is there another place where I should also remove it?

Below are some additional screenshots of the settings I forgot to add yesterday:
General tab:
![General_tab](https://user-images.githubusercontent.com/41524566/156072692-567db491-e9e6-4d00-9677-f1321b5fb5d7.png)

Connection tab:
![Connection_tab](https://user-images.githubusercontent.com/41524566/156072673-5ee36356-d96a-4875-9868-3df9f3839853.png)

Gonna lower the Scan Connection to 2 or 1 and see if that also helps.

**Comment by thomas694** [comment]
You can find `LastId` inside the blog file (e.g. gyurism.tumblr) in the Index folder.

Yes, that's the right location of the log file, except if you would use portable mode, then it's, of course, inside the current folder. If no errors occur, not much should be written there. But in the case here it doesn't seem to be much helpful.

With the settings posted, it should be enough to just remove the blog from the list. But to be sure, you can manually check that there are no files with "gyurism" in the name below the Index folder.

But looking on the date of the files, it might be a problem with old structures as already mentioned in the other post.

**Comment by Lumerica** [comment]
I saw the blog file under the portable folder '\Blogs\Index' the LastId was 0.
However the last date modified was earlier than I ran last. I have moved all the blog files just to make sure the folder is empty to start anew.

I made 3 test runs, 2 was with SVC crawler 1 was with API.
The first run, there was still an old blog file in the directory because I was not aware of at first.
The other two runs there was no blog file in that directory created neither did it seems like photos are missing in the directory at first look.

The first run I did about 10+ hours ago with SVC, I emptied the download folder and choose a different destination. The total downloads was 2100 something out of 2700 something, I am sorry I did not remember the exact number as I wanted to try again after. But what was funny is that there were some pictures downloaded that was not included in the 2nd run.

The 2nd run, which was about less roughly 5 hours ago with SVC, the app shows that there are 7536 under the column 'Downloaded Files' and 'Number of Downloads' says 9154.

The 3rd run also ran about 5 hours ago was with API however there are 9111 Downloaded Files and Number of Downloads is 9154.

I am thinking that its related to the way how the crawler handles the posts between SVC and API? But according to the tooltip, the SVC will check for higher resolution files than the API. So I would prefer to use SVC.

By the way is it normal that even after these runs that the Blogs\Index directory to be empty?

**Comment by Lumerica** [comment]
I just tried with a different blog:

https://esraa7.tumblr.com

With SVC as crawler the number of downloads is around ~2700 Downloaded files and ~2600 Number of downloads, with API its 16833 Downloads files and 15650 number of downloads.

But it should be the otherway around right? that I should be able to download more with SVC cause it requires login plus higher resolution. I would assume that with SVC if it cannot find a higher resolution files it will just download the best that is available with the settings I have set.

**Comment by thomas694** [comment]
> By the way is it normal that even after these runs that the Blogs\Index directory to be empty?

If you changed the download location, then the blog files are inside an Index folder in the new location.

> I am thinking that its related to the way how the crawler handles the posts between SVC and API? But according to the tooltip, the SVC will check for higher resolution files than the API. So I would prefer to use SVC.

The tooltip is a bit outdated. Normally both API and SVC crawler should be able to download the same resolutions now. API and SVC are two different ways how the data is offered and hence how TumblThree can download it. But in general, both should download roughly the same.
The only difference is that Tumblr offers some blogs only in SVC way and you need to be logged in, while normal blogs are offered publicly in both ways. Normally TumblThree is choosing the correct crawler automatically on blog addition.

> But it should be the otherway around right?

As we already found out, there is some error (in SVC crawler) that needs to be addressed.

**Comment by thomas694** [comment]
The issue has been fixed and closed. You can still comment. Feel free to ask for reopening the issue if needed.
